core/src/Serial/Kokkos_Serial.hpp

```diff
@@ -30,6 +30,7 @@ static_assert(false,
 
 #include <cstddef>
 #include <iosfwd>
+#include <iterator>
```

Comment:
why is this needed here?

Reply:
On Windows with NVCC the implicit includes are different exposing a missing include I beleive it was for `std::back_inserter`

Reply:
See https://github.com/kokkos/kokkos/pull/6659#discussion_r1421613190.